src/components/textbox/textbox.pw.tsx

```diff
@@ -0,0 +1,1244 @@
+
+  test("should render with positionedChildren prop", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextboxComponentWithPositionedChildren />);
+
+    const textboxParent = await textbox(page).evaluateHandle(
+      (element: Element) => {
+        return element.parentElement;
+      }
+    );
+
+    const buttonExistsInTextboxParent = await page.evaluate(
+      (parentDomElement) => {
+        const element = parentDomElement?.querySelectorAll(
+          `[data-component="button"]`
+        ).length;
+        return element;
+      },
+      textboxParent
+    );
+
+    expect(buttonExistsInTextboxParent).toBeTruthy();
+  });
+
+  ([
+    ["flex", 399],
+    ["flex", 400],
+    ["block", 401],
+  ] as [string, TextboxProps["adaptiveLabelBreakpoint"]][]).forEach(
+    ([displayValues, breakpoint]) => {
+      test(`should render with ${displayValues} label alignment when the adaptiveLabelBreakpoint prop is ${breakpoint} with a set viewport of 400`, async ({
+        mount,
+        page,
+      }) => {
+        await page.setViewportSize({ width: 400, height: 300 });
+        await mount(<TextboxComponent adaptiveLabelBreakpoint={breakpoint} />);
+
+        const label = getDataElementByValue(page, "label");
+        const labelParentParent = await label.evaluateHandle(
+          (element: Element) => {
+            return element.parentElement?.parentElement;
+          }
+        );
+
+        const displayValue = await getStyle(
+          (labelParentParent as unknown) as Locator,
+          "display"
+        );
+
+        expect(displayValue).toEqual(displayValue);
+      });
+    }
+  );
+
+  test("should render with labelId prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent id={CHARACTERS.STANDARD} />);
+
+    const label = getDataElementByValue(page, "label");
+
+    await expect(label).toHaveAttribute("id", `${CHARACTERS.STANDARD}-label`);
+    await expect(label).toHaveAttribute("for", `${CHARACTERS.STANDARD}`);
+  });
+
+  testData.forEach(([fieldHelp]) => {
+    test(`should render with fieldHelp prop as ${fieldHelp}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent fieldHelp={fieldHelp} />);
+
+      await expect(fieldHelpPreview(page)).toHaveText(fieldHelp);
+    });
+  });
+
+  test("should render with formattedValue prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent formattedValue={CHARACTERS.STANDARD} />);
+
+    await expect(textboxInput(page)).toHaveAttribute(
+      "value",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  (["add", "filter", "play"] as TextboxProps["inputIcon"][]).forEach((icon) => {
+    test(`should render with inputIcon as ${icon}`, async ({ mount, page }) => {
+      await mount(<TextboxComponent inputIcon={icon} />);
+
+      const inputIcon = getDataElementByValue(page, "input-icon-toggle");
+      await expect(textbox(page).locator(inputIcon)).toBeVisible();
```

Comment:
```suggestion
      await expect(inputIcon).toBeVisible(;
```

**question** (non-blocking): maybe I'm missing something but this seems much simpler. The only difference is this way doesn't need the icon to be inside the textbox div, whereas yours does - but does this actually matter? (I would think not but I'm not sure.)

Reply:
Technically no, but I don't see the harm in keeping this in either. I might just leave it as i think its easier to understand then just locating the inputIcon